students/elaine_x/session05/mailroom3.py

```diff
@@ -0,0 +1,130 @@
+'''
+##########################
+#Python 210
+#Session 05 - Mailroom Part 3
+#Elaine Xu
+#Feb 12,2019
+###########################
+'''
+import sys
+import datetime
+import operator
+
+def send_a_thankyou():
+    '''send a thankyou note to donor'''
+    # creating a list of donors - lower case
+    donor_list = [donor.lower() for donor in donor_db]
+    #print(donor_list) #for testing
+    #prompt for a full name
+    while True:
+        fullname = input("Enter full name of the donor (type 'list' to show all donor names): ")
+        if fullname.lower() == 'list':
+            for donor in donor_db:
+                print(donor)
+        elif fullname.lower() not in donor_list:
+            print(f'{fullname} does not exist in current donor data, '
+                  f'adding {fullname} to the donor data.')
+            #prompt for a donation amount
+            while True:
+                try:
+                    donation_amount = float(input("Enter the donation amount: "))
+                    break
+                except ValueError:
+                    print("Donation amount has to be a number, try again.")
```

Comment:
Need a `continue` here?

Reply:
is the 'continue' necessary? I think the 'while' loop will keep running till user inputs something valid.